cms/static/js/views/import.js

```diff
@@ -2,241 +2,306 @@
+
+                    if (current.stage !== 0) {
+                        current.state = STATE.IN_PROGRESS;
+                        displayFeedbackList();
+
+                        self.pollStatus(current.stage);
                     }
-                );
+                });
             },
 
             /**
-             * Entry point for server feedback. Makes status list visible and starts
-             * sending requests to the server for status updates.
-             * @param {string} url The url to send Ajax GET requests for updates.
+             * Starts the importing process.
+             * Makes status list visible and starts showing upload progress.
+             *
+             * @param {string} fileName The name of the file to import
+             * @param {string} fileUrl The full URL to use to query the server
+             *     about the import status
              */
-            startServerFeedback: function (url){
-                this.stopGetStatus = false;
-                getStatus(url, 1000, 0);
+            start: function (fileName, fileUrl) {
+                current.state = STATE.IN_PROGRESS;
+
+                file.name = fileName;
+                file.url = fileUrl;
+
+                initEventListeners();
+                storeImport();
+                displayFeedbackList();
+                updateFeedbackList();
             },
 
             /**
-             * Give error message at the list element that corresponds to the stage
-             * where the error occurred.
-             * @param {int} stageNo Stage of import process at which error occurred.
-             * @param {string} msg Error message to display.
+             * Fetches the previous stored import
+             *
+             * @return {JSON} the data of the previous import
              */
-            stageError: function (stageNo, msg) {
-                this.stopGetStatus = true;
-                var all = $('ol.status-progress').children();
-                // Make all stages up to, and including, the error stage 'complete'.
-                var prevList = all.slice(0, stageNo + 1);
-                _.map(prevList, function (elem){
-                    $(elem).
-                        removeClass("is-not-started").
-                        removeClass("is-started").
-                        addClass("is-complete");
-                    updateCog($(elem), false);
-                });
-                var message = msg || gettext("There was an error with the upload");
-                var elem = $('ol.status-progress').children().eq(stageNo);
-                if (!elem.hasClass('has-error')) {
-                    elem.removeClass('is-started').addClass('has-error');
-                    elem.find('p.copy').hide().after("<p class='copy error'>" + message + "</p>");
-                }
+            storedImport: function () {
+                return JSON.parse($.cookie('lastfileupload'));
```

Comment:
@AkA84 nit: magic constant - might make sense to actually use a constant instead of a string.